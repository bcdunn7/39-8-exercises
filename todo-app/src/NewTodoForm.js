import React, { useState } from 'react';

const NewTodoForm = ({ addTodo }) => {

    const INITIAL_STATE = {
        name: ""
    }

    const [formData, setFormData] = useState(INITIAL_STATE);

    const handleChange = e => {
        const { name, value } = e.target;
        setFormData(fData => ({
            ...fData,
            [name]: value
        }));
    };

    const handleSubmit = e => {
        e.preventDefault();
        const { name } = formData;
        addTodo({ name })
        setFormData(INITIAL_STATE);
    }

    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="name">New Todo:</label>
            <input
                type="text"
                id="name"
                name="name"
                value={formData.name}
                onChange={handleChange}
            />
            <button>Add!</button>
        </form>
    )
}

export default NewTodoForm;