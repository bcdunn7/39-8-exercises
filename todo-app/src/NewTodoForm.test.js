import React from "react";
import { render } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

test("it renders without crashing", function() {
	render(<NewTodoForm/>);
});

test("it matches snapshot", () => {
    const { asFragment } = render(<NewTodoForm/>);
    expect(asFragment()).toMatchSnapshot();
})