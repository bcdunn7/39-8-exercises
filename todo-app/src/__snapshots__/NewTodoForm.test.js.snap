// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="name"
    >
      New Todo:
    </label>
    <input
      id="name"
      name="name"
      type="text"
      value=""
    />
    <button>
      Add!
    </button>
  </form>
</DocumentFragment>
`;
